pkg/operator/pool/controller.go

```diff
@@ -208,13 +208,32 @@ func (p *PoolSpec) erasureCode() *ErasureCodedSpec {
 	return nil
 }
 
-func (p *PoolSpec) Validate() error {
+func (p *PoolSpec) Validate(context *clusterd.Context, namespace string) error {
 	if p.replication() != nil && p.erasureCode() != nil {
 		return fmt.Errorf("both replication and erasure code settings cannot be specified")
 	}
 	if p.replication() == nil && p.erasureCode() == nil {
 		return fmt.Errorf("neither replication nor erasure code settings were specified")
 	}
+
+	// validate the failure domain if specified
+	if p.FailureDomain != "" {
+		crush, err := ceph.GetCrushMap(context, namespace)
+		if err != nil {
+			return fmt.Errorf("failed to get crush map. %+v", err)
+		}
+		found := false
+		for _, rule := range crush.Types {
```

Comment:
we are iterating over types in the crush map, not rules, right? this variable may be misleading then

Reply:
agreed, will rename